client.go

```diff
@@ -67,7 +119,7 @@ func (c *Client) CreateDownchannel(accessToken string) (<-chan *Message, error)
 
 // Do posts a request to the AVS service's /events endpoint.
 func (c *Client) Do(request *Request) (*Response, error) {
-	body := &bytes.Buffer{}
+	body := NewBlockingReader()
```

Comment:
The `blockingReader` type has quite a bit going on inside of it for the purposes of avoiding `bytes.Buffer`'s EOF on empty. Do you think `io.Pipe()` would work here? It's used for purposes like these where you need an intermediate in-memory buffer.

Here's a quick example:

```
out, in := io.Pipe()
writer := multipart.NewWriter(in)
// ...
// Send the request to AVS.
req, err := http.NewRequest("POST", EventsURL, out)
```

Inside the goroutine I think you need to call `out.Close()` when done copying, or `out.CloseWithError(err)` if an error occurs.